.github/workflows/docker-push.yml

```diff
@@ -0,0 +1,33 @@
+name: Build and Push Subgraph Deployer Docker Image
+
+on:
+  push:
+    tags:
+      - '*' # Trigger on any tag it is pushed by release workflow
```

Comment:
Does this mean on any branch?

Reply:
It's mean on any tags

Reply:
Actually, we want to only build tagged docker images on tags coming from `main` (for `vX.Y.Z` tags) and `release` (for `vX.Y.Z-rc*`) branches.

Reply:
Check if the docker build

Reply:
It's build only the docker when we add a tag in branch. And we can trigger an manual deploy with get the latest `vX.Y.Z` tag and add `+dev+sha`.